aea/configurations/base.py

```diff
@@ -152,7 +150,7 @@ def read_all(self) -> List[Tuple[str, T]]:
         return [(k, v) for k, v in self._items_by_id.items()]
 
 
-class PublicId(object):
+class PublicId(JSONSerializable):
```

Comment:
now `PublicId` is JSONSerializable, mainly due to serialization issues (see later changs in the error handler).